pytest: Temporary Working Directory

ax3l opened the pull request:
Tests can generate temporary output and directories, which might clash and/or need cleaning between tests.

This changes the current working directory to a unique directory per test.

ax3l committed:
pytest: Temporary Working Directory

Tests can generate temporary output and directories, which
might clash and/or need cleaning between tests.

This changes the current working directory to a unique
directory per test.

sayerhs commented on tests/conftest.py:
```diff
@@ -20,20 +20,21 @@
+                # "abort_on_out_of_gpu_memory=1",
+            ]
+        )
     yield
     amr.finalize()
```
Recommend that these two lines are also inside the `with` block

ax3l replied:
I was googling about `with` context and `yield` combinations and got nervous about what I saw :sweat_smile:

ax3l commented on tests/conftest.py:
```diff
@@ -20,20 +20,21 @@
+            ]
+        )
     yield
     amr.finalize()
```
Bravely trying my best:
```suggestion
        yield
        amr.finalize()
```

ax3l committed:
All the yielded results in the cwd context

ax3l commented:
Hm, this actually fails my tests in https://github.com/ECP-WarpX/impactx/pull/424 ... maybe because it works now (and cannot find input files)?

sayerhs commented:
> Hm, this actually fails my tests in [ECP-WarpX/impactx#424](https://github.com/ECP-WarpX/impactx/pull/424) ... maybe because it works now (and cannot find input files)?

What is the error? How are you creating the input files?

ax3l committed:
Set basepath for input files (later)

Not yet used, can be used in test files as
```py
from conftest import basepath
```

ax3l commented:
Ah, now managed to pass their path before going into the per-test directory :)
